### ISSUE TITLE
Horizontal Review: Accessibility self test

### ISSUE THREAD

**Issue by iherman** [w3c/did-core#105]
To initiate an Accessibility Review, a group is supposed to look at the [APA checklist](http://w3c.github.io/apa/fast/checklist.html) to see what is relevant for this group.

The main sections of the survey are below. I checked those items that, in my view, _are not relevant for DID and DID documents_  (therefore, as far as the check is concerned, we are o.k.). In case of doubt I have either added a comment to explain why I do not believe it is of any relevance for us, or I also listed the specific sub-items. Again, I checked those that I think we are fine with (mostly because they are not relevant).

- [X] If technology allows visual rendering of content
- [X] If technology provides author control over color
- [X] If technology provides features to accept user input
- [X] If technology provides user interaction features
- [X] If technology defines document semantics
  - Comment: the questions refer to 'semantics' of a readable (e.g., HTML) document, like identifying regions of text of understandings of acronyms, abbreviations, etc.
- [X] If technology provides time-based visual media (see also the Media Accessibility Checklist)
- [X] If technology provides audio
- [X] If technology allows time limits
- [x] If technology allows text content
  - [x] Authors can define non-text alternatives for text content.
  - [x] Authors can define non-text alternatives for non-text content.
- [X] If technology creates objects that don't have an inherent text representation
    - [X] There is a mechanism to create short text alternatives that label the object.
    - [X] There is a mechanism to create extended text alternatives for fallback content.
    - [X] Text alternatives can be semantically "rich" e.g., with page structure, text style, hyperlinks, etc.
  
  I think these questions are not relevant for our case; DID documents are inherently managed and understood by machines, and not by humans. (The explanatory text in the test refer to things like form controls, labels, etc.)
- [X] If technology provides content fallback mechanisms, whether text or other formats
- [X] If technology provides visual graphics
- [X] If technology provides internationalization support
- [X] If technology defines accessible alternative features
- [X] If technology provides content directly for end-users
- [X] If technology defines an API
- [X] If technology defines a transmission protocol
  - [X] Use of the does not cause any aspect of the content, including metadata which could contain important accessibility information, to be removed
  - [X] It is possible to use third-party accessibility enhancement services while using the protocol.

**Comment by iherman** [comment]
The same question arises as in https://github.com/w3c/did-core/issues/104#issuecomment-553303192 with the 9th item above: does a DID document include natural language text. If the answer is 'no' then that item can also be check-marked.

**Comment by msporny** [comment]
> The same question arises as in #104 (comment) with the 9th item above: does a DID document include natural language text. If the answer is 'no' then that item can also be check-marked.

In general, it shouldn't because that quickly leads to DLTs running afoul of GDPR.

There is a possibility, since this is Linked Data, to point elsewhere that does store data in a GDPR "Right to be Forgotten" compliant way. In either case, since we're using JSON-LD, all of the mechanisms that we can use for a11y that exist there, we can use here.

So, I think we're ok, and will have to see if the a11y folks have specific things they're concerned about, and if so, we should have specific answers for them. Given that the Verifiable Credentials data model and syntax passed a11y review, I expect this one to as well as it's very close in design and implementation.

**Comment by iherman** [comment]
> There is a possibility, since this is Linked Data, to point elsewhere that does store data in a GDPR "Right to be Forgotten" compliant way. In either case, since we're using JSON-LD, all of the mechanisms that we can use for a11y that exist there, we can use here.

Yes, of course. I could also imagine that one decides to use extra terms in a different vocabulary to add a pointer to one's CV to the DID document. I presume that should allowed; after all, as you say, this is Linked Data. But that is all right: _we_ do not control those extra vocabularies and their namespaces, i.e., the possible i18n problems would have to be settled for those who defined that particular vocabulary.

I let the @brentzundel and @burnburn to decide when should this particular issue be considered as settled, and I can then forward the reference to this issue to the I18N group.

**Comment by iherman** [comment]
The latest editorial round on the Core (after the F2F meeting of Amsterdam) means that we will have to check these self-tests again. Once that is done, we should connect the WAI people to start the official horizontal review.

**Comment by iherman** [comment]
Discussed on 2020-04-14: when the current §3 and §6 gets filled in (which means that the final structure of the document is in place), we can go out to get the horizontal review done.

**Comment by burnburn** [comment]
@iherman can you please update status?

**Comment by iherman** [comment]
@burnburn https://github.com/w3c/did-core/issues/313 has to be closed by @xfq to declare victory; I presume that can be done only when #319 is indeed merged.

**Comment by burnburn** [comment]
@iherman yes, we came to the same conclusion on our call.

**Comment by iherman** [comment]
@burnburn, @brentzundel,  Per https://github.com/w3c/did-core/issues/104#issuecomment-649187234 and https://github.com/w3c/did-core/issues/104#issuecomment-649583144 we are done with I18N. We will use this issue as a reference for our CR transition call; we can leave it open to make the transition easier, but we can also call it. As you prefer.

**Comment by iherman** [comment]
Oops... We have been mixing up the issues! #105 (the present one, is on A11y, not I18n)...

The following comments should be disregarded:

- https://github.com/w3c/did-core/issues/105#issuecomment-648626945
- https://github.com/w3c/did-core/issues/105#issuecomment-648677596
- https://github.com/w3c/did-core/issues/105#issuecomment-649587277

(@brentzundel @burnburn do you prefer to delete the issue comments?)

Thanks to @xfq for noticing this.

**Comment by iherman** [comment]
Got the feedback from Janina Sajka (reproduced here with permission):

>I have completed reading through the DID specification as of today's
date:
>
> https://www.w3.org/TR/did-core/
>
> I note and appreciate your responses to our (nascent) accessibility self-check instrument:
>
> `https://github.com/w3c/did-core/issues/105`
>
> I would agree there are no accessibility concerns in this specification. Moreover, it does appear there are excellent opportunities to apply this specification to various accessibility use cases--admittedly a different matter.
>
> My apologies for taking so long to complete this review. I do expect we will clear our various tracking checks during today's APA call at Noon (Boston).
>
> Best,
>
> Janina

@brentzundel @burnburn : just for the records, this means that the A11y horizontal review is now complete. We should leave this issue open, referring to it when we make our report for the CR transition.

**Comment by brentzundel** [comment]
Leaving this issue open until CR

**Comment by msporny** [comment]
This is done, we're keeping this open until we make the request to go to CR.

**Comment by brentzundel** [comment]
Horizontal review is complete, closing
